scalafix/rules/src/main/scala/rsc/rules/RscCompat.scala

```diff
@@ -11,39 +11,39 @@ import rsc.rules.syntax._
 import scala.meta._
 import scala.meta.contrib._
 import scala.meta.internal.{semanticdb => s}
-import scalafix.internal.util._
-import scalafix.internal.v0._
-import scalafix.rule._
-import scalafix.syntax._
+import scalafix.internal.v1._
+import scalafix.lint.Diagnostic
+import scalafix.patch.Patch
+import scalafix.v1.{Configuration, Rule}
 import scalafix.util.TokenOps
-import scalafix.v0._
+import scalafix.v1.SemanticDocument
+import scalafix.v1._
 
-case class RscCompat(legacyIndex: SemanticdbIndex, config: RscCompatConfig)
-    extends SemanticdbRule(legacyIndex, "RscCompat") {
-  def this(legacyIndex: SemanticdbIndex) = {
-    this(legacyIndex, RscCompatConfig.default)
-  }
+case class RscCompat(config: RscCompatConfig)
+    extends SemanticdbRule( "RscCompat") {
+  def this() = this(RscCompatConfig.default)
 
-  override def init(config: Conf): Configured[Rule] = {
-    config
+  override def withConfiguration(config: Configuration): Configured[Rule] = {
+    config.conf
       .getOrElse("rscCompat", "RscCompat")(RscCompatConfig.default)
-      .map(RscCompat(legacyIndex, _))
+      .map(RscCompat(_))
   }
 
-  override def fix(ctx: RuleCtx): Patch = {
-    val targets = collectRewriteTargets(ctx)
-    targets.map(ascribeReturnType(ctx, _)).asPatch
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    val targets = collectRewriteTargets(doc)
+    targets.map(ascribeReturnType(doc, _)).asPatch
   }
 
-  private case class RewriteTarget(
+  case class RewriteTarget(
```

Comment:
Why are this class and the methods below changed to public?

Reply:
The class needs to be public to work with the v1 config loader. The others were public so I could test them more easily. I created adhoc tests for them to get the rule working then removed the tests as they were mostly for my learning and not otherwise useful.

Reply:
> The class needs to be public to work with the v1 config loader. 
Why?

Reply:
Make it private and it fails to compile with a inaccessible error. I will reproduce later and add the message here.